bin/ch-test

```diff
@@ -297,40 +305,102 @@ test_run () {
     set +e
 }
 
-test_make_auto () {
-    local img
-
-    printf "finding images compatible with test settings ...\n"
-    for i in $images $examples; do
-        if (test_make_check_image "$i"); then
-            echo "found: $i"
-            targets+=( "$i" )
-        fi
-    done
-
-    printf '\n'
+test_make () {
+    local bat_file
+    local tag
+    printf "finding artifacts compatible with %s test settings ...\n" "$phase"
     case $phase in
     build)
+        for i in $images $examples; do
+           if (test_make_check_image "$i"); then
+               echo "found: $i"
+               build_targets+=( "$i" )
+           fi
+        done
+        printf '\n'
         printf 'generate build_auto.bats ...\n'
-        "$make_auto" build "${targets[@]}" > "${TMP_}/build_auto.bats"
+        test_make_auto "$phase" "${build_targets[@]}" > "${TMP_}/build_auto.bats"
         printf 'ok\n\n'
         ;;
     run)
+        for i in "${CH_TEST_TARDIR}"/*{.tar.gz,.sqfs}; do
+            if [[ -e $i ]]; then
+                artifact=${i##*/}
+                artifact=${artifact%%.*}
+                for j in $images $examples; do
+                    if [[ $(test_make_tag_from_path "$j") == "$artifact" ]]; then
+                        if (test_make_check_image "$j"); then
+                            echo "found: $i"
+                            run_targets+=( "$j" )
+                        fi
+                    fi
+                done
+            fi
+        done
+        printf '\n'
```

Comment:
For each artifact found in `--pack-dir` look for a test artifact that produces a matching tag. If we find one, then check the image for exclusion conditions.

Reply:
Make this a comment. :)